fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashActivity.java

```diff
@@ -0,0 +1,232 @@
+package com.github.dedis.popstellar.ui.digitalcash;
+
+import android.annotation.SuppressLint;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.IdRes;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.utility.ActivityUtils;
+import com.github.dedis.popstellar.utility.Constants;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+
+import java.util.function.Supplier;
+
+import dagger.hilt.android.AndroidEntryPoint;
+
+/**
+ * Activity for the digital cash
+ */
+@AndroidEntryPoint
+public class DigitalCashActivity extends AppCompatActivity {
+    private DigitalCashViewModel mViewModel;
+    public static final String TAG = DigitalCashActivity.class.getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.digital_cash_main_activity);
+        mViewModel = obtainViewModel(this);
+        setupNavigationBar();
+
+
+        setTheIntent();
+        // Subscribe to "open home"
+        openHomeEvent();
+        // Subscribe to "open history"
+        openHistoryEvent();
+        // Subscribe to "open send"
+        openSendEvent();
+        // Subscribe to "open receive"
+        openReceiveEvent();
+        // Subscribe to "open issue"
+        openIssueEvent();
+        // Subscribe to "open receipt"
+        openReceiptEvent();
+        mViewModel.openHome();
+    }
+
+
+    public void setTheIntent() {
+        if (getIntent().getExtras() != null) {
+            mViewModel.setLaoId(getIntent().getExtras().getString(Constants.LAO_ID_EXTRA, ""));
+            mViewModel.setLaoName(getIntent().getExtras().getString(Constants.LAO_NAME, ""));
+            mViewModel.setRollCallId(getIntent().getExtras().getString(Constants.ROLL_CALL_ID, ""));
```

Comment:
The usage of the Constant class is an improvement !
Though once again the Lao id was probably the only thing needed to retrieve the rest of the intent information

Reply:
yes as discussed in the above, can be refactored but maybe can wait for after the presentations.